### PR TITLE
Migrated CloudWatch Insights to new CoreAddons model

### PR DESCRIPTION
**Pull request by 5herlocked**
*Issue #, if available:* #971, #948

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/cloud-watch-insights/index.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+import {conflictsWith, createNamespace as makeNamespace, supportsALL} from "../../utils";
 import {CoreAddOn, CoreAddOnProps} from "../core-addon";
 import {ebsCollectorPolicy} from "./iam-policy";
-import {IManagedPolicy, ManagedPolicy} from "aws-cdk-lib/aws-iam";
-import {KubernetesVersion, ServiceAccount} from "aws-cdk-lib/aws-eks";
+import {KubernetesVersion} from "aws-cdk-lib/aws-eks";
 
 const versionMap: Map<KubernetesVersion, string> = new Map([
-  [KubernetesVersion.V1_29, "v1.3.1-eksbuild.1"],
+  [KubernetesVersion.V1_29, "v1.4.0-eksbuild.1"],
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:10]: Default version for `1.29` is `v1.5.5-eksbuild.1`. Likewise please check for other 4 versions.

**Review by elamaran11 — COMMENTED** [review]
@5herlocked Minor feedback on `versionMap`, @shapirov103 double check from your end. This is for issue #971

File: lib/addons/cloud-watch-insights/index.ts (hunk 1 above)
Comment (shapirov103) [reply lib/addons/cloud-watch-insights/index.ts:10]: +1

**Review by shapirov103 — COMMENTED** [review]
@5herlocked  have you validated that 1/ addon creates correct service account and 2/ you can set the version explicitly, or auto as well ?

**Comment by 5herlocked** [comment]
> @5herlocked have you validated that 1/ addon creates correct service account and 2/ you can set the version explicitly, or auto as well ?

I have validated both

File: test/cloudwatch-insights.test.ts [hunk 2]
```diff
@@ -43,14 +44,44 @@ const cloudwatchInsightsAddOnProps = {
...
     const app = new cdk.App();
 
-    const blueprint = blueprints.EksBlueprint.builder()
+    const blueprint = await blueprints.EksBlueprint.builder()
+      .account("123456789012").region('us-east-2')
+      .version(KubernetesVersion.V1_29)
+      .addOns(new blueprints.CloudWatchInsights())
+      .buildAsync(app, 'cloudwatch');
+
+    const template = Template.fromStack(blueprint);
```
Comment (shapirov103) [inline test/cloudwatch-insights.test.ts:56]: github code quality issue here. please run `make lint`

**Review by shapirov103 — COMMENTED** [review]
@5herlocked please address the minor comment, before submitting please make sure you run all commands specified https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/internal/readme-internal.md#submitting-pull-requests.

```
npm i
make build
make lint
make run-test
cdk list
```

**Review by shapirov103 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**